### PR TITLE
Fix dynamic capabilities and implement did change watched files

### PR DESCRIPTION
**Pull request by CGNonofr**

### CODE REVIEW

File: src/language-client.ts [hunk 1]
```diff
@@ -70,7 +72,6 @@ export class LanguageClient implements Disposable {
   }))
 
   private currentDocuments = new Map<string, TextDocument>()
-  // private currentDocumentMasters = new Map<string, TextDocuments<TextDocument>>()
```
Comment (samuel-olivier) [inline src/language-client.ts:73]: do we want to keep this comment ?

Comment (CGNonofr) [reply src/language-client.ts:73]: No, why wouldn't we?

Comment (samuel-olivier) [reply src/language-client.ts:73]: Then we should remove it ;)

Comment (CGNonofr) [reply src/language-client.ts:73]: it is!?

Comment (samuel-olivier) [reply src/language-client.ts:73]: Damn I don't know how to read a diff anymore

**Review by samuel-olivier — APPROVED** [review]

**Pull request merged**